build.gradle

```diff
@@ -312,11 +256,9 @@ task copyStylelintConfiguration(type: Copy) {
 }
 
 task setupIntellijStaticAnalysis {
-    description 'Sets up the static analysis plugins in the IntelliJ project'
-    dependsOn copyIntelliJProjectPluginsSettings
-    finalizedBy copyStylelintConfiguration
+    description 'Sets up the static analysis plugins in the IntelliJ project.'
+    dependsOn copyIntelliJProjectPluginsSettings, copyStylelintConfiguration
```

Comment:
`copyStylelintConfiguration` currently fails on Windows due to [this](https://discuss.gradle.org/t/error-with-a-copy-task-on-windows/1803), though it's a pre-existing issue in `master`.

Reply:
That is really strange. I think I run into that thread before and did not believe the person because it just does not make sense Gradle does not allow for that and that thread is really old, and this would cause an exception which I'm sure I managed to avoid.

Though I think I saw this thread because I vaguely remember running into this issue, saw some file locking issue and just unlocked my files (I think?)

Though if you are running into it again, maybe we have to reconsider this or something.

Reply:
Ok I have confirmed that Gradle really decided not to fix this issue lol.
https://issues.gradle.org/browse/GRADLE-3002
Looks like we have to litter our codebase everywhere with this caveat whenever we copy to the project root.

Reply:
This is weird because I'm pretty sure `gradle.template.properties` is in root...

Reply:
Looks like `createConfigs` is not a copy task, but calls `copy` manually, which seems to work? I believe we can use something similar for `copyStylelintConfiguration` without resorting to calling `ant`.

Reply:
I'm pretty sure they run into the same bug but `createConfigs` specifically filter each file so the `.gradle` folder doesn't get locked.  I was always wondering why the code have to be so complex, it's just that the author didn't explain why he wrote the way he did that?

I prefer Ant here because the code is more readable because even if we don't use Ant we have to label the caveat otherwise it would be like today that we have learned why the author wrote the code the way he did.

Not to mention this is recommended by the Gradle team.

Reply:
If you're referring to the author of `createConfigs`, that's me, and no I don't know about any of these. All I cared about back then was that it works in Travis/Mac OS/Windows (AppVeyor was not up yet back then).

Reply:
@wkurniawan07 oh LOL so I guessed you indirectly avoided the bug? Sorry I edited my comment again so do you think we should use something like you did, but I think we still have to add comments to everywhere that copies to the project root.

Reply:
@whipermr5 have devised a much better solution that is simple and readable, that is similar to what @wkurniawan07 have implemented. Thus this is resolved

I must have understood the behavior of copy method wrongly, my apologies. I am suspecting that it is along the lines of not requiring the computation of hashes.